.github/workflows/win_ci.yml

```diff
@@ -0,0 +1,260 @@
+      CHECK_OPTIONS: 'CSHARPOPTIONS=-platform:x64'
+      SWIGLANG: ${{ matrix.SWIGLANG }}
+      COMPILER: ${{ matrix.COMPILER }}
+      INSTALL: ${{ matrix.INSTALL }}
+      VER: ${{ matrix.VER }}
+      OS: ${{ matrix.os }}
+
+# cl.exe:
+#  https://learn.microsoft.com//cpp/build/reference/compiler-options
+#  /EHc  extern "C" defaults to nothrow.
+#  /EHs  Enable C++ exception handling (no SEH exceptions).
+#  /W3   Warning level.
+#  /WX   Treat warnings as errors.
+#  https://learn.microsoft.com//cpp/build/reference/linker-options
+#  cl.exe pass linker:
+#  /VERBOSE:LIB        Outputs progress messages during the link process.
+#  /NODEFAULTLIB:lib   Ignore library 'lib'
+
+    steps:
+    - name: Machine Info
+      shell: cmd
+      run: |
+          systeminfo | findstr /B /C:"OS Name" /B /C:"OS Version"
+
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        show-progress: true
+
+    - name: Install CCache
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: ${{ matrix.os || 'windows-2022' }}-${{ matrix.COMPILER || 'msvc' }}
+
+    - name: Install NuGet Packages
+      if: ${{ env.COMPILER == '' }}
+      shell: cmd
+      run: |
+          rem 'nuget build for .NET: https://www.nuget.org/packages'
+          nuget install PCRE2 -OutputDirectory C:\Tools
+          nuget install boost -OutputDirectory C:\Tools
+
+    - name: Setup MSVC
+      if: ${{ env.COMPILER == '' || env.SWIGLANG == 'csharp' }}
+      uses: TheMrMilchmann/setup-msvc-dev@v3
+      with:
+        arch: x64
+
+    - name: Prepare Environment
+      shell: bash
+      run: |
+          uname --all
+          # Tabs are used for here-documents indentation!
+          # YAML indentation is done with spaces only!
+          # Rest of bash code beside here-documents uses spaces as well :-)
+
+          if [[ "$COMPILER" = "gcc" ]]; then
+            PCRE2_PATH=/c/msys64/usr
+            PCRE2_LD=-lpcre2-8
+
+            if [[ "$SWIGLANG" = "python" ]]; then
+              MORE_MSYS_PKGS=python
+            fi
+
+            cat <<- EOF >> $GITHUB_ENV
+          		MORE_MSYS_PKGS=gcc pcre2-devel mingw-w64-x86_64-boost $MORE_MSYS_PKGS
+          		LIBS_CCACHE=-luserenv
+          		BOOST_PATH=/c/msys64/mingw64
+          	EOF
+
+          else # COMPILER: cccl wrapping MSVC
+
+            curl --retry 15 -s -L https://github.com/swig/cccl/raw/master/cccl > /usr/bin/cccl
+            chmod +x /usr/bin/cccl
+            cp -p /usr/bin/cccl /c/msys64/usr/bin/cccl
+
+            PCRE2_PATH=$(ls -d /c/tools/PCRE2*)
+            PCRE2_LD=$PCRE2_CCCL_LD
+
+            cat <<- EOF >> $GITHUB_ENV
+          		CXX=/usr/bin/cccl
+          		CC=/usr/bin/cccl
+          		BOOST_PATH=$(ls -d /c/tools/boost*)/lib/native
+          	EOF
+
+            if [[ "$SWIGLANG" = "python" ]] && [[ -n "$VER" ]]; then
+              PY3VER=$(cygpath -w $(ls -d /c/hostedtoolcache/windows/Python/$VER*)/x64)
```

Comment:
I see the versions of Python listed at https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md. Out of interest, did you have to reverse engineer how to set this path or is it documented somewhere else?

Reply:
The pathes are not document. Also the Docker files/VM that create the image are difficult to find.
You may call it reverse engineer, I simply did a search in the image, that is why I run so many test on GitHub.

Reply:
Sounds similar to what I had to do for configuring the appveyor builds!